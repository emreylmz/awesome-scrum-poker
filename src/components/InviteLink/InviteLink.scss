@import "../../styles/colors";

.invite-link {
  display: flex;
  align-items: flex-end;
  font-size: 12px;
  align-self: flex-start;
  flex: 1;

  &__info {
    flex: 1;
  }

  &__url {
    flex: 2;
    display: inline-block;
    color: $secondary-color;
    background-color: $text-color;
    margin: 10px;
    font-weight: bold;
    font-size: 12px;
    border-radius: 5px;
    padding: 10px;
  }
}
