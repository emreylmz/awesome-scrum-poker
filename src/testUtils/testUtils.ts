export const emptyTestFunc = (): void => void 0
