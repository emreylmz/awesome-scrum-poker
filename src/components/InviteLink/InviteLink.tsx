import React from 'react'
import { Render } from '../../types'
import './InviteLink.scss'

interface Props {
  link: string
}

class InviteLink extends React.PureComponent<Props> {
  render(): Render {
    const { link } = this.props
    const url = `${window.location.origin}${link}`
    return (
      <div className={'invite-link'}>
        <span className={'invite-link__info'}>Please share link of developers panel to the teammates</span>
        <div className={'invite-link__url'}>{url}</div>
      </div>
    )
  }
}

export type InviteLinkProps = Props
export default InviteLink
