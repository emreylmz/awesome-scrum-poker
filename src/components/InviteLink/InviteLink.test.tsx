import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import '../../testUtils/setupTests'
import InviteLink, { InviteLinkProps } from './InviteLink'

type DoneCallback = jest.DoneCallback

const defaultLink = '/default-lint'
const getWrapper = (props?: InviteLinkProps): ShallowWrapper => {
  return shallow(<InviteLink link={defaultLink} {...props} />)
}
describe('Invite Link UI', (): void => {
  it('should render correctly', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it('should show invite link', (done: DoneCallback): void => {
    const testUrl = `${window.location.origin}${defaultLink}`
    const wrapper = getWrapper()
    const linkWrapper = wrapper.find('.invite-link__url')
    expect(linkWrapper.text()).toEqual(testUrl)
    done()
  })
})
