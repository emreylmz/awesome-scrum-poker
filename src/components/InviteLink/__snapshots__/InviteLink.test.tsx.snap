// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invite Link UI should render correctly 1`] = `ShallowWrapper {}`;
